server/routes/updateGoal/updateGoalController.ts

```diff
@@ -41,10 +41,11 @@ export default class UpdateGoalController {
   submitUpdateGoalForm: RequestHandler = async (req, res, next): Promise<void> => {
     const { prisonNumber, goalReference } = req.params
     const updateGoalForm: UpdateGoalForm = { ...req.body }
+    const updateStepForm: UpdateStepForm = { ...req.body }
     req.session.updateGoalForm = updateGoalForm
 
```

Comment:
You want to remove line 44; then here (before the validation), you want to do something like:
```
if (updateGoalForm.action === 'delete-step') {
    // work out which step to remove .... not sure how to do that yet, needs thought... 🤔 
   // remove the step from the array updateGoalForm.steps
   // Redirect back to the Update Goal page
    return res.redirect(`/plan/${prisonNumber}/goals/${goalReference}/update`)
}
```

Reply:
If we check for deleting steps before the validation check, then you can do away with your change on line 48, and we wont have the problem where you cant delete a step you've accidentlally added with no/blank data

Reply:
Based on what I've just suggested about form action on `UpdateGoalForm` I think this part of the code would be
```
    // Remove the desired step on the action delete step
    if (updateStepForm.action.startsWith('delete-step-')) {
      // Get the step index inbetween the 2 characters [ ] from the action value
      const stepIndex = updateStepForm.action.match(/\[(.*?)\]/)[1]

      // Remove the desired step from the array
      updateGoalForm.steps.splice(stepIndex, 1)

      // Re-sequence the step array so that all the step's stepNumber fields are sequential starting from 1
      ....
      ....

      // Redirect back to the Update Goal page with named anchor taking the user to the edit and remove steps section
      return res.redirect(`/plan/${prisonNumber}/goals/${goalReference}/update#edit-and-remove-steps`)
    }
```